src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,537 @@
+                }
+                if (left < Size.Width && top < Size.Height)
+                {
+                    _lcd.SetCursorPosition(left, top);
+                }
+                CursorLeft = left;
+                CursorTop = top;
+            }
+        }
+
+        /// <summary>
+        /// Write text to display.
+        /// </summary>
+        /// <remarks>
+        /// There are only 256 characters available. There are chip variants
+        /// with different character sets. Characters from space ' ' (32) to
+        /// '}' are usually the same with the exception of '\', which is a
+        /// yen symbol on some chips '¥'.
+        /// </remarks>
+        /// <param name="text">Text to be displayed.</param>
+        public void Write(string text)
+        {
+            text = text.Replace("\r\n", "\n"); // Change to linux format only, so we have to consider only this case further
+
+            List<string> lines = text.Split("\n", StringSplitOptions.None).ToList();
+            FindLineWraps(CursorLeft, lines);
+            for (int i = 0; i < lines.Count; i++)
+            {
+                string line = lines[i];
+                string currentLine = line;
+
+                int remainingChars = Math.Min(currentLine.Length, Size.Width - CursorLeft);
+                if (remainingChars > 0)
+                {
+                    lock (_lock)
+                    {
+                        WriteCurrentLine(currentLine.Substring(0, remainingChars));
+                    }
+                }
+
+                if (i != lines.Count - 1)
+                {
+                    // Insert newline except after the last text segment.
+                    NewLine();
+                }
+            }
+
+        }
+
+        /// <summary>
+        /// Asynchronously writes text to the display.
+        /// See <see cref="Write(string)"/> for limitations.
+        /// Only one write operation (synchronous or asynchronous) should be executed at once, otherwise the execution order 
+        /// is undefined. 
+        /// </summary>
+        /// <param name="text">Text to write</param>
+        /// <returns>A task object</returns>
+        public Task WriteAsync(string text)
+        {
+            return Task.Factory.StartNew(() =>
+            {
+                Write(text);
+            });
+        }
+
+        /// <summary>
+        /// Writes text, starting at the given line number. 
+        /// This will overwrite the text in the given line, filling up with spaces, if needed. Line wrapping is disabled, so this method will write exactly one line. 
+        /// </summary>
+        /// <param name="lineNumber">0-based index of the line to start</param>
+        /// <param name="text">Text to insert. No newlines supported.</param>
+        /// <exception cref="ArgumentException">The string contains newlines.</exception>
+        public void Write(int lineNumber, string text)
```

Comment:
I'm not sure this should be called `Write` - before I read the description I thought this will insert on that line and wasn't sure if it will insert at start or end but it actually replaces which suggest the name should be clarified (i.e. ReplaceLine)

Reply:
You are right, this behaves differently to the Write methods. Changed to ReplaceLine.